File: LeetCode/2352. Equal Row and Column Pairs/solution.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        cc = Counter()
        n = len(grid)
        ans = 0 
        for i in range(n):
            cc[tuple([grid[i][j] for j in range(n)])] += 1
        for j in range(n):
            ans += cc[tuple([grid[i][j] for i in range(n)])]
        return ans 

File: LeetCode/2352. Equal Row and Column Pairs/solution.cpp
class Solution {
    const int P = 1000007;
    const int MOD = 998244353;

public:
    int equalPairs(vector<vector<int>>& grid) {
        int n = grid.size();
        vector<long long> R(n), C(n);
        for (int i = 0; i < n; i++) for (int j = 0; j < n; j++) R[i] = (R[i] * P + grid[i][j]) % MOD;
        for (int j = 0; j < n; j++) for (int i = 0; i < n; i++) C[j] = (C[j] * P + grid[i][j]) % MOD;

        int ans = 0;
        for (int i = 0; i < n; i++) for (int j = 0; j < n; j++) if (R[i] == C[j]) ans++;
        return ans;
    }
};


作者：TsReaper
链接：https://leetcode.cn/circle/discuss/sWfXxC/view/gEN00n/
来源：力扣（LeetCode）
著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。